### PR TITLE
Sprint 06/render messages tab

### PR DESCRIPTION
**Pull request by tarasbashuk**

### CODE REVIEW

**Review by MaksymSimonov — APPROVED** [review]

File: client/src/reducers/chat.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    case CHATS_RECEIVED:
+      return { ...state, chats: payload, chatsLoading: false }
+
+    case MESSAGES_RECEIVED:
+      return { ...state, chatMessages: payload, messagesLoading: false }
+
+    case SEND_MESSAGE:
+      /* eslint-disable */
+      const newKey = ++state.keyForRender
```
Comment (RostyslavStepanchuk) [inline client/src/reducers/chat.js]: Somebody mutates state over here )))

File: client/src/reducers/chat.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      const newChatMessages = state.chatMessages
+      newChatMessages.push(payload)
+      /* eslint-enable */
+      return { ...state, chatMessages: newChatMessages, keyForRender: newKey }
```
Comment (RostyslavStepanchuk) [inline client/src/reducers/chat.js]: As far as I understand eslint doesn't like variables declaration within case, why not to try then:
  return { ...state,
              chatMessages: state.chatMessages.concat(payload), 
              keyForRender: state.keyForRender + 1 }
             }

File: src/main/java/com/socialmedia/controller/ChatMessageController.java [hunk 3]
```diff
@@ -28,8 +31,12 @@ public ChatMessageController(ChatMessageMapper chatMessageMapper) {
   @GetMapping("/{chatId}")
   public ResponseEntity<List<ChatMessageDtoOut>> getAllMessagesForChat(
             @PathVariable Long chatId,
-            @PageableDefault(sort = { "id" }, direction = Sort.Direction.DESC) Pageable pageable) {
+            @PageableDefault(sort = { "id" }, direction = Sort.Direction.ASC) Pageable pageable) {
```
Comment (RostyslavStepanchuk) [inline src/main/java/com/socialmedia/controller/ChatMessageController.java:34]: Are you sure about ascending? This is pageable, so we'll always receive the very first messages of our chat, when asking for first page. But in chat you'd want to see the last ones

**Review by RostyslavStepanchuk — APPROVED** [review]

File: client/src/reducers/chat.js (hunk 1 above)
Comment (tarasbashuk) [reply client/src/reducers/chat.js]: Agree, will be change soon

File: client/src/reducers/chat.js (hunk 2 above)
Comment (tarasbashuk) [reply client/src/reducers/chat.js]: Thanks a lot! I did almost the same but instead concat I used push... Yesterday was a tough day ))

File: src/main/java/com/socialmedia/controller/ChatMessageController.java (hunk 3 above)
Comment (tarasbashuk) [reply src/main/java/com/socialmedia/controller/ChatMessageController.java:34]: I changed it cause I render messages from top to bottom. But I didn't reckon that I receiving only last 10. I`ll find out how to manage it.

**Pull request merged**